### PR TITLE
Training new pocket representation 

### LINKED ISSUE #103: Pocket representation
# Pocket-only representation
To make sure we dont have to build entirely seperate datasets for the pocket representation, this implementation should just get index positions for our binding pocket and then just apply a mask to the original graph (similar to how it is done with the [`dropout_node`](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dropout_node) function in `pytorch_geometric`).

**Task list:**
- [x] #109
- [x] #110
- [x] #123
- [ ] Train models and evaluate

## [KLIFS Database](https://klifs.net/details.php?structure_id=3834)
This is used by KBDNet to get the binding pockets for davis and kiba "pocket is 85 residues long".

The sequence given from KLIFS is not contiguous and only contains relevant pocket residues. 
- So to match it up with our structures we need to do a sequence alignment and then **get the index positions of those residues**.

After we get our list of index positions for the binding pocket AA we can modify our existing graph by applying a mask 
- This way we don't need to build entirely separate databases, and can just apply the mask before inference

## Getting pockets for Kiba
1. Using **UniProt ID** we can get the pocket from the KLIFS database with the `/kinase_ID` API.
	1. For example: https://klifs.net/api/kinase_ID?kinase_name=O00141&species=HUMAN returns: image

## Getting pockets for davis:
Same as for kiba, but we use the raw **Gene Name** code (need to remove any mutation or phosphorylation information): `ABL1(F317I)p` -> `ABL1`
	1. For example: https://klifs.net/api/kinase_ID?kinase_name=ABL1&species=HUMAN returns: 
image

However **for mutated genes**  we must be careful with the sequence alignment, and must follow the following procedure to get the right amino acid index positions:
1. Reverting the mutation
3. Perform alignment 
4. Extract index positions

then we just use these positions for our mask on the original (mutated) graph.

### PR DESCRIPTION
**Pull request by jyaacoub**
resolves #103

### CODE REVIEW

**Review by jyaacoub — COMMENTED** [review]
ltgm
